### DISCUSSION TITLE
Feature Request: Change the first day of the week

### DISCUSSION THREAD

**Discussion by ajszoke** [iSoron/uhabits#421] [Feature Requests] [answered]
I like to consider a week started on Monday, as is the standard in Europe and other places, instead of on Sunday. So when viewing a specific habit's page, in the "History" and "Frequency" grids, it would be nice for the first row of the grids to be Monday instead of Sunday. So, I would like there to be a setting to select what the first day of the week should be, and update the grids accordingly.

**Comment by hoseinit** [comment]
I Agree
Also in Persian Calendar, Saturday is the first day of the week. It would be great if we could change the starting day of the week accordingly.
Thanks

**Comment by iSoron** [comment]
The app is already supposed to select the first day of the week according to your locale. Could you guys please let me know (i) what locale (language & time zone) are you using? (ii) what was your expected first day of the week? (iii) what day of the week did the app select instead?

**Comment by ajszoke** [comment]
Thanks for my reply. My locale is EST/English, so by that logic my app correctly set its first day to Sunday. However, my personal preference is Monday as I mentioned, and I am able to override my locale default first day in almost every other app I use that uses calendars. I would appreciate similar functionality in this app.

**Comment by hoseinit** [comment]
Thanks for answering, 
My Locale is IRST and my preferred first day of the week is Saturday.

**Comment by sciamano** [comment]
@ajszoke : why do you want to go that complicated way and change preferences in every app instead of just fixing you locale settings once and get the correct first day (and more) in all places automatically? The system locales are not just some funny stuff, they are here for a reason and for quite long time.

**Comment by ajszoke** [comment]
That setting is not available on its own on Android. The only way I could force that would be to pick another city with a shared time zone, all of which are Spanish speaking cities. 
null

**Comment by rick-nu** [comment]
It would be great if it's possible to change the "first day of the week" in the uhabits app settings.

I live in the Netherlands, yet the app starts at Sunday instead of Monday. Probably because my primary language is English instead of Dutch.

**Comment by xavier-pardonnet** [comment]
Same problem. My language setting is en-us because that's my working language, and I want to keep using the US spelling, but I live in the NL, so I would like my start off the week to be Monday (it's Sunday atm)

**Comment by woj-tek** [comment]
AFAIR you can set locale to en_GB and that will only affect the UI. Spell-checking is separated so they won't clash. 

Arguably setting en-gb locale is the most sensible option out there for everyone wanting device in English without US silliness related to dates… And it works for all apps in the system.

**Comment by xavier-pardonnet** [comment]
I found a way to change the locale to a more sensible one on https://www.ctrl.blog/entry/android-en150-localization.html
As much as it seem it would solve a lot of problems, it's still quite involved, and probably more so than than adding a simple "Week start on Monday" option to the app, as so many have, imho.

**Comment by woj-tek** [comment]
Or, you can use British locale, which virtually achieve almost the same thing instead of asking that every app include this switch...

**Comment by iSoron** [comment, marked as answer]
It's now possible to change the first day of the week in the settings. These changes will ship in Loop 1.8.
